### ISSUE TITLE
Attack reproduce (Table 3)

### ISSUE THREAD

**Issue by YunSeoHwan** [facebookresearch/audioseal#79]
Hi!

Thank you for releasing the AudioSeal code. I am currently reproducing the detection performance on a wide range of audio editing operations. While most of the attacks were successfully reproduced, I encountered an issue with the part highlighted in red — the detection performance was significantly lower.
I used the attack code exactly as provided in the AudioSeal repository and conducted the evaluation on the [VoxPopuli](https://huggingface.co/datasets/facebook/voxpopuli) validation set.
Do you happen to know what might be causing this discrepancy?

<img width="1119" alt="Image" src="https://github.com/user-attachments/assets/2ec84aa5-8145-406d-b7de-09a19e1770d9" />

**Comment by pierrefdz** [comment]
Hi! Yes indeed it looks weird.

1/ For the audios, could you check the SR and durations when you load, and make sure that if you save some audios after the attack there is no error in the audios.
2/ Could you share the attack parameters as well?
3/ Could you share how you calibrate the scores? I remember that there was a big difference if we used the scores from all the attacks as negative scores, or did it separately for each attack.

I'll have a more thorough look on my side when I have the time.

**Comment by YunSeoHwan** [comment]
Thank you so much for taking the time to check everything.

1) I only saved the metrics after the attack, so I haven’t been able to check the details yet. I’ll look into that and get back to you. Additionally, when I downloaded the VoxPopuli validation dataset, I received 18K samples. Among them, around 8K were longer than 10 seconds, and 10K were 10 seconds or shorter. The paper mentions **“10k ten-seconds audios from our
VoxPopuli validation set,”** but I was wondering — which subset is this referring to?

2) I set the attack parameters according to section D.2 (E) of the paper. The sampling rate was set to 16KHz.

3) I used accuracy_score, roc_curve, and auc from sklearn.metrics. Below is the specific code I used:
``` python
metrics_start_time = time.time()
metrics_data = []
for attack_name, data in results.items():
    if not data['labels']:
        print(f"Warning: No results recorded for attack '{attack_name}'. Skipping metrics calculation.")
        metrics = {'attack': attack_name, 'accuracy': np.nan, 'tpr': np.nan, 'fpr': np.nan, 'auc': np.nan, 'samples_processed': 0}
        metrics_data.append(metrics)
        continue

    labels = np.array(data['labels'])
    confidences = np.array(data['confidences'])
    predictions = np.array(data['predictions'])
    samples_processed = len(labels)

    try:
        accuracy = accuracy_score(labels, predictions)

        if len(np.unique(labels)) < 2:
            roc_auc = np.nan
        else:
            fpr, tpr, thresholds = roc_curve(labels, confidences)
            roc_auc = auc(fpr, tpr)

        tn = np.sum((labels == 0) & (predictions == 0))
        fp = np.sum((labels == 0) & (predictions == 1))
        fn = np.sum((labels == 1) & (predictions == 0))
        tp = np.sum((labels == 1) & (predictions == 1))

        tpr_at_threshold = tp / (tp + fn) if (tp + fn) > 0 else np.nan
        fpr_at_threshold = fp / (fp + tn) if (fp + tn) > 0 else np.nan

        metrics = {
            'attack': attack_name,
            'accuracy': accuracy,
            'tpr': tpr_at_threshold,
            'fpr': fpr_at_threshold,
            'auc': roc_auc,
            'samples_processed': samples_processed
        }
        metrics_data.append(metrics)
```

4) I also had one more question — which dataset did you use to evaluate audio/speech quality?

Thank you again for generously spending your time. I truly appreciate it!
